docs/spec/index.md

```diff
@@ -1,54 +1,67 @@
-| ops_location_lon | Longitude | Required | The longitude of the operational location. |
+| **File Name** | **Type** | **Description** |
+| --- | --- | --- |
+| trips_supplement.txt | Supplement | Supplements and modifies GTFS [trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt) with deadheads and other non-public trip information.|
+| stops_supplement.txt | Supplement | Supplements and modifies GTFS [stops.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stopstxt) with internal stop locations, waypoints, and other non-public stop information.|
+| stop_times_supplement.txt | Supplement | Supplements and modifies GTFS [stop_times.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stop_timestxt) with non-public times that trips stop at locations, and related information. |
+| routes.txt | Supplement | Supplements and modifies GTFS [routes.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#routestxt) with internal route identifiers and other non-public route identification. |
+| runs_pieces.txt | TODS-Specific | Defines daily personnel schedules within a feed. |
+| run_events.txt | TODS-Specific| Defines other scheduled activities to be performed by a member of personnel during a run. |
 
-### deadhead_times.txt
+The use of the `_supplement` standard to modify other GTFS files remains experimental at this time.
+
+
+## `_supplement` Files
+
+### Structure & Evaluation
+
+The fields in all `_supplement` files match those defined in the corresponding file's [GTFS specification](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md).
+
+Each row in a `_supplement` file is used to modify the corresponding file in GTFS and should be evaluated as follows:
+
+1. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is equal to `1`, remove the row from the GTFS file.
+2. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is not defined or not equal to `1`, set or update (in the corresponding GTFS file) the value of any non-blank fields defined in the `_supplement` file's row to the `_supplement`'s value.
+3. If the row's Primary key is NOT defined in the corresponding GTFS file, add the row to the corresponding GTFS file.
+
+In other words, where a primary key matches, the row is either removed or any non-empty values in the row are used to *update* the corresponding GTFS values. Where a primary key match does not exist, the entire row is added.
+
+> _**Implications and Guidance:**_
+> - To explicitely blank a field, delete the row entirely using the `TODS_delete` field and then re-add the row with the desired values subsequent thereto.
+> - If a row contains defined values besides the Primary Key and a `TODS_delete` value of `1`, the row shall be removed and other values in that row will be ignored.
```

Comment:
This all seems correct, but I had a really hard time reading it. Part of that is cuz the rules are kind of complicated, but maybe a rewrite could make it clearer?

Some more specific suggestions:
- The main idea of adding+updating+deleting rows gets obscured by all the details. Perhaps there should be a simple explanation first that glosses over some details, followed by a less-readable full list of rules?
- The behavior of blank fields is barely mentioned, but will be a very important thing that people use all the time. Perhaps it deserves more weight, especially in the readable-but-imprecise guide.
- "Primary Key" may require a bit more explanation. It's not something that matters too much for most people's use of GTFS, so people might not be familiar with it. Maybe just a link to [the GTFS docs about it](https://gtfs.org/schedule/reference/#dataset-attributes), or a "The columns that make up a file's Primary Key are listed with each file in its docs" explanation or something.
- Typo: "explicitely" -> "explicitly".

Reply:
Good points!
- I split "Structure" into a new section to include more of the core idea before getting into the evaluation process.
- Blank field detail added to "Implications & Guidance"
- Primary Key described and linked as suggested
- Typo corrected